god-mode.el

```diff
@@ -351,6 +351,76 @@ Members of the `god-exempt-major-modes' list are exempt."
            (throw 'disable t))
          (setq preds (cdr preds)))))))
 
+(defun god-execute-with-current-bindings (&optional called-interactively)
+    "Execute a single command from God mode, preserving current keybindings.
+
+This command activates God mode temporarily, and deactivates God
+mode as soon as the next command is run.  Prefix arguments do not
+count as commands for this purpose, and do not cause God mode to
+exit.  Moreover, any prefix argument that exists at the time of
+this command's invocation is passed along to the next command.
+
+Unlike normal use of God mode, this command makes available all
+keybindings that were active at the time of its invocation,
+including keybindings that are normally invisible to God mode,
+such as those in `emulation-mode-map-alists' or text overlay
+properties.  This makes it suitable for use with packages like
+Evil that utilize such higher-priority keymaps.  (See Info
+node `(elisp)Active Keymaps' for technical details on keymap
+precedence.  For an alternative to this command, check out the
+evil-god-state package, available on MELPA.)
+
+This command has no effect when called from within God mode.
+
+For interactive use only.  CALLED-INTERACTIVELY is a dummy
+parameter to help enforce this restriction."
+    (interactive "d")
+    (if called-interactively
+        (unless god-local-mode
+          (message "Switched to God mode for the next command ...")
+          (letrec ((caller this-command)
+                   (buffer (current-buffer))
+                   (temp-map (make-composed-keymap
+                              (cons god-local-mode-map (current-active-maps))))
+                   (cleanup
+                    (lambda ()
+                      ;; Perform cleanup in original buffer even if the command
+                      ;; switched buffers
+                      (with-current-buffer buffer
+                        (unwind-protect (god-local-mode 0)
+                          (remove-hook 'post-command-hook post-hook)))))
+                   (post-hook
+                    (lambda ()
+                      (unless (and
+                               (eq this-command caller)
+                               ;; If we've entered the minibuffer, this implies
+                               ;; a non-prefix command was run, even if
+                               ;; `this-command' has not changed.  For example,
+                               ;; `execute-extended-command' behaves this way.
+                               (not (window-minibuffer-p)))
+                        (funcall kill-temp-map))))
+                   (kill-temp-map
```

Comment:
For readability, `kill-temp-map` should be bound before `post-hook` in the `letrec` form.